src/battle-animations.ts

```diff
@@ -626,10 +626,10 @@ class BattleScene {
 				if (speciesOverage && speciesTable.includes(species)) {
 					for (const sidebarIcon of sidebarIcons) {
 						if (side.pokemon[sidebarIcon[1]!].getBaseTemplate().baseSpecies === species) {
-							sidebarIcon[0] = 'pokemon-illusion';
+							if (this.gen < 8) sidebarIcon[0] = 'pokemon-illusion';
```

Comment:
For another thing, do you know what this does? This makes it so that if Species Clause is in effect but there are two Pokémon of the same species on the field, one of them is marked as an Illusion. This seems obviously correct.